sql_runner/snowflake_target.go

```diff
@@ -188,3 +211,36 @@ func stringify(row [][]byte) []string {
 	}
 	return line
 }
+
+// Blocking function to poll for the true status of a query which didn't return a result.
+func pollForQueryStatus(sft SnowFlakeTarget, queryId string) error {
+	// Get the snoflake driver and open a connection
+	sfd := sft.Client.Driver()
+	conn, err := sfd.Open(sft.Dsn)
+	if err != nil {
+		return errors.Wrap(err, "Failed to open connection to poll for query result.")
+	}
+	// Poll Snowflake for actual query status
+	for {
+		qStatus, err := conn.(sf.SnowflakeConnection).GetQueryStatus(context.Background(), queryId)
+
+		switch {
+		case err != nil && strings.Contains(err.Error(), "[RUNNING]"):
+			// If there's an error, but it contains `[RUNNING]`, the query hasn't yet completed.
+			break
+		case err != nil:
+			// Any other error is genuine, return the error.
+			return err
+		case qStatus != nil && qStatus.ErrorCode == 0:
+			// A non-nil qStatus means the query completed. If the ErrorCode field is 0, we have no error.
+			return nil
+		case qStatus != nil:
+			// If qStatus is non-nil but has a non-zero error code, return the relevant info as an error.
+			return errors.New(qStatus.ErrorMessage)
```

Comment:
Yeah from my reading of things it shouldn't be possible to hit this case, but if that's the case I don't understand why there's an `ErrorMessage` field in the `qStatus` object, so I handled it just in case my understanding is wrong and there actually is a case where we can get a status with an error in it.